dev_docs/building_blocks.mdx

```diff
@@ -0,0 +1,140 @@
+---
+id: kibBuildingBlocks
+slug: /kibana-dev-docs/building-blocks
+title: Building blocks
+summary: Consider these building blocks when developing your plugin.
+date: 2021-02-24
+tags: ['kibana','onboarding', 'dev', 'architecture']
+---
+
+When building a plugin in Kibana, there are a handful of architetural "building blocks" you can use. Some of these building blocks are "higher-level", 
+and some are "lower-level". High-level building blocks come 
+with many built-in capabilities, require less maintenance, and evolve new feature sets over time with little to no
+ manual changes required. When developers use high-level building blocks, new features are exposed consistently, across all of Kibana, at the same time.
+  On the downside, they are not as flexible as our low-level building blocks. 
+ 
+ Low-level building blocks
+ provide greater flexibility, but require more code to stitch together into a meaningful UX. This results in higher maintenance costs and greater UI/UX variability
+ across Kibana.
+
+ For example, if an application is using <DocLink id="kibBuildingBlocks" section="index-patterns" text="Index Patterns"/> and 
+ <DocLink id="kibBuildingBlocks" section="search-source" text="Search Source"/>, 
+ their application would automatically support runtime fields. If the app is instead using the
+ lower-level <DocLink id="kibBuildingBlocks" section="search-strategy" text="Search Strategy"/>, additional work would be required.
+
+Armed with this knowledge, you can choose what works best for your use case!
+
+# Application building blocks
+
+## UI components
+
+The following high-level building blocks can be rendered directly into your application UI.
+
+### Query Bar
+
+The data plugin provides a high-level Query Bar component that comes with support for Lucene, KQL, Saved Queries, 
+and <DocLink id="kibBuildingBlocks" section="index-patterns" text="Index Patterns"/>.
+
+If you would like to expose the ability to search and filter on Elasticsearch data, the Query Bar provided by the
+<DocLink id="kibDataPlugin" text="Data plugin"/> 
+ is your go-to building block.
+
+**Github labels**: `Team:AppServices`, `Feature:QueryBar`
+
+### Dashboard Embeddable
+
+Add a Dashboard Embeddable directly inside your application to provide users with a set of visualizations and graphs that work seamlessly
```

Comment:
No links in the system to send people to, to read about this component. No one is actually using it like this yet, but we do have one example plugin that shows how to add a by-value dashboard embeddable. The SIEM team is looking for by-reference support before using this for their overview page.